### PR TITLE
[Program: GCI] Create user dashboard endpoint [Mentorship - Backend]

### LINKED ISSUE #415: Create user dashboard endpoint
# Description
Create endpoint which gathers all information at ones: pending requests for any relation send invitations for any relations and tasks.

# Acceptance Criteria
Create new endpoint which returns object containing:
- list of received invitation as a mentee (separate list for each state: PENDING, ACCEPTED,  REJECTED, CANCELLED, COMPLETED)
- list of send invitation as a mentee (separate list for each state)
- list of received invitation as a mentor (separate list for each state)
- list of send invitation as a mentor (separate list for each state)
- separate lists of done and todo tasks 
- All list have to be ordered by creation date. 
- Relation lists need to contain the user's display name, profile image URL and creation date. Tasks lists need to contain all tasks details.

# Definition of Done:
- [x] list of received invitation as a mentee (separate list for each state: PENDING, ACCEPTED,  REJECTED, CANCELLED, COMPLETED)
- [x] list of send invitation as a mentee  (separate list for each state: PENDING, ACCEPTED,  REJECTED, CANCELLED, COMPLETED)
- [x] list of received invitation as a mentor  (separate list for each state: PENDING, ACCEPTED,  REJECTED, CANCELLED, COMPLETED)
- [x] list of send invitation as a mentor  (separate list for each state: PENDING, ACCEPTED,  REJECTED, CANCELLED, COMPLETED)
- [x] separate lists of done and todo tasks 
- [x] All list have to be ordered by creation date. 
- [x] Relation lists need to contain the user's display name, profile image URL and creation date. Tasks lists need to contain all tasks details.

# May be fixed by:
#378 
#379 
#380

### PR DESCRIPTION
**Pull request by asawesome07**
### Description

Added GET /dashboard endpoint

Fixes #415 

### Type of Change:
- Code
- User Interface

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
I tested this using `python -m unittest discover tests`

Here's a screenshot
c1

Here's how it looks in swagger
Capture

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: app/api/dao/user.py [hunk 1]
```diff
@@ -418,3 +422,91 @@ def get_user_statistics(user_id):
...
+        self.mentee_id = relation.mentee_id
+        self.action_user_id = relation.action_user_id
+        self.response = {
+            'id': relation.id,
+            'action_user_id': relation.action_user_id,
+            'mentor_id': relation.mentor.id,
+            'mentee_id': relation.mentee.id,
+            'mentor_name': relation.mentor.name,
+            'mentee_name': relation.mentee.name,
+            'mentor_profile_picture_URL': relation.mentor.photo_url,
```
Comment (annabauza) [inline app/api/dao/user.py]: I would keep the convention of small url. I would put mentor and mentee as sub dictionary, so you could have helper method for parsing user. What do you think?

**Review by annabauza — CHANGES_REQUESTED** [review]
Please have a look at comments I've made.

File: app/api/dao/user.py (hunk 1 above)
Comment (asawesome07) [reply app/api/dao/user.py]: fixed the small url thing

Comment (asawesome07) [reply app/api/dao/user.py]: could you please explain what you mean by putting mentor and mentee in sub dictionary?

**Review by annabauza — APPROVED** [review]

**Pull request merged**